queries.sql

```diff
@@ -62,3 +62,72 @@ SELECT MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight FROM animals;
 
 -- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
 SELECT CAST(AVG(escape_attempts) AS DECIMAL(5,2)) AS average_escape_attempts FROM animals WHERE extract(YEAR FROM date_of_birth) BETWEEN '1990' AND '2000';
+
+-- JOIN QUERIES
+-- What animals belong to Melody Pond?
+SELECT 
+an.name AS animal_name,
+ow.full_name AS owner_name 
+FROM animals an 
+INNER JOIN owners ow
+ON ow.id = an.owners_id 
+WHERE ow.full_name = 'Melody Pond';
+
+-- List of all animals that are pokemon (their type is Pokemon).
+SELECT 
+an.name AS animal_name, 
+sp.name AS species 
+FROM animals an 
+INNER JOIN species sp
+ON sp.id = an.species_id
+WHERE sp.name = 'Pokemon';
+
+-- List all owners and their animals, remember to include those that don't own any animal.
+SELECT
+ow.full_name as owner_name,
+an.name as animal_name
+FROM owners ow
+LEFT JOIN animals an
+ON ow.id = an.owners_id
+
+-- How many animals are there per species?
+SELECT 
+sp.name AS species_name,
+COUNT(an.name) AS animal_count
+FROM animals an 
+INNER JOIN species sp
+ON sp.id = an.species_id
+GROUP BY sp.name;
+
+-- List all Digimon owned by Jennifer Orwell.
+SELECT 
+ow.full_name as owner_name,
+sp.name as species_name,
+an.name as animal_name
+FROM animals an 
+INNER JOIN owners ow
+ON ow.id = an.owners_id
+INNER JOIN species sp
+ON sp.id = an.owners_id
+WHERE ow.full_name = 'Jennifer Orwell';
```

Comment:
- [ ] Nice job on this query, fetching animals owned by _Jennifer Orwell_, however, we're actually looking for the animals of the species type `Digimon` owned by _Jennifer Orwell_. Hence, let's improve our query so that it's similar to the below:
```sql
   SELECT animals.* FROM animals
   LEFT JOIN owners
   ON animals.owner_id = owners.id
   LEFT JOIN species
   ON animals.species_id = species.id
   WHERE owners.full_name = 'Jennifer Orwell'
   AND species.name = 'Digimon';
```

Reply:
Okey Thanks